web-wallet/src/routes/(welcome)/setup/restore/+page.svelte

```diff
@@ -28,12 +34,19 @@
 	/** @type {string[]} */
 	let mnemonicPhrase = [];
 
+	const collectSettings = getKey("userId");
+	const userId = collectSettings($settingsStore);
+
 	$: if (showPasswordSetup) {
 		password = showPasswordSetup ? password : "";
 	}
 </script>
 
-{#if !tosAccepted}
+{#if !notice && userId.length !== 0}
```

Comment:
As we said yesterday `userId` might not exist, in theory, if the user comes from an older version of the web-wallet.
We can just check that `userId` is not a "falsy" value here (as you did in the login page with `currentAddress`.